src/ResourceManagement/ResourceManager/Microsoft.Azure.Management.ResourceManager.Fluent/Region.cs

```diff
@@ -1,75 +1,135 @@
+        public static readonly Region US_EAST2 = new Region("eastus2");
+        public static readonly Region US_NORTH_CENTRAL = new Region("northcentralus");
+        public static readonly Region US_SOUTH_CENTRAL = new Region("southcentralus");
+        public static readonly Region US_WEST_CENTRAL = new Region("westcentralus");
+        public static readonly Region CANADA_CENTRAL = new Region("canadacentral");
+        public static readonly Region CANADA_EAST = new Region("canadaeast");
+        public static readonly Region BRAZIL_SOUTH = new Region("brazilsouth");
+        #endregion
+
+        #region Europe
+        public static readonly Region EUROPE_NORTH = new Region("northeurope");
+        public static readonly Region EUROPE_WEST = new Region("westeurope");
+        public static readonly Region UK_SOUTH = new Region("uksouth");
+        public static readonly Region UK_WEST = new Region("ukwest");
+        #endregion
+
+        #region Asia
+        public static readonly Region ASIA_EAST = new Region("eastasia");
+        public static readonly Region ASIA_SOUTHEAST = new Region("southeastasia");
+        public static readonly Region JAPAN_EAST = new Region("japaneast");
+        public static readonly Region JAPAN_WEST = new Region("japanwest");
+        public static readonly Region AUSTRALIA_EAST = new Region("australiaeast");
+        public static readonly Region AUSTRALIA_SOUTHEAST = new Region("australiasoutheast");
+        public static readonly Region INDIA_CENTRAL = new Region("centralindia");
+        public static readonly Region INDIA_SOUTH = new Region("southindia");
+        public static readonly Region INDIA_WEST = new Region("westindia");
+        #endregion
+
+        #region China
+        public static readonly Region CHINA_NORTH = new Region("chinanorth");
+        public static readonly Region CHINA_EAST = new Region("chinaeast");
+        #endregion
+
+        #region German
+        public static readonly Region GERMANY_CENTRAL = new Region("germanycentral");
+        public static readonly Region GERMANY_NORTHEAST = new Region("germanynortheast");
+        #endregion
+
+        #region Government Cloud
+        public static readonly Region GOV_US_VIRGINIA = new Region("usgovvirginia");
+        public static readonly Region GOV_US_IOWA = new Region("usgoviowa");
+        #endregion
+
+        public static IReadOnlyCollection<Region> Values
+        {
+            get
+            {
+                return regions.Values as IReadOnlyCollection<Region>;
+            }
+        }
+
+        public string Name
+        {
+            get; private set;
+        }
+
+        private Region(string name)
+        {
+            Name = name.ToLowerInvariant();
+            regions.Add(Name, this);
+        }
+
+        public static Region Create(string name)
+        {
+            name = sWhitespace.Replace(name, "").ToLowerInvariant();
```

Comment:
used @"\s+" to take care of all whitespace char types including tab and newline.